mbed_greentea/mbed_greentea_cli.py

```diff
@@ -85,7 +85,42 @@ def print_version(verbose=True):
     if verbose:
         print version
     return version
-
+    
+def create_filtered_test_list(ctest_test_list, opts):
```

Comment:
I think you can just have function declared like this:
```python
def create_filtered_test_list(ctest_test_list, test_by_names, skip_test):
```
and replace in function body:
* ```opts.skip_test``` with ```skip_test``` and
* ```opts.test_by_names``` with ```opts.test_by_names```.